diff -ruN -x .svn fit/cpp/fit_buffer_encode.cpp libfit/cpp/fit_buffer_encode.cpp
--- fit/cpp/fit_buffer_encode.cpp	2012-02-01 20:24:36.000000000 -0800
+++ libfit/cpp/fit_buffer_encode.cpp	2012-04-18 11:16:57.292653519 -0700
@@ -13,7 +13,7 @@
 // Profile Version = 2.0Release
 // Tag = $Name: AKW2_000 $
 ////////////////////////////////////////////////////////////////////////////////
-
+#include <string.h>
 
 #include "fit_buffer_encode.hpp"
 #include "fit_crc.hpp"
diff -ruN -x .svn fit/cpp/fit_decode.cpp libfit/cpp/fit_decode.cpp
--- fit/cpp/fit_decode.cpp	2012-02-01 20:24:36.000000000 -0800
+++ libfit/cpp/fit_decode.cpp	2012-04-23 17:08:01.840214737 -0700
@@ -166,7 +166,7 @@
    state = STATE_FILE_HDR;
    lastTimeOffset = 0;
 
-   file.seekg(0, std::ios::beg);
+   //file.seekg(0, std::ios::beg);
 }
 
 Decode::RETURN Decode::ReadByte(FIT_UINT8 data)
@@ -211,7 +211,7 @@
                if ((data & FIT_PROTOCOL_VERSION_MAJOR_MASK) > (FIT_PROTOCOL_VERSION_MAJOR << FIT_PROTOCOL_VERSION_MAJOR_SHIFT))
                {
                   std::ostringstream message;
-                  message << "FIT decode error: Protocol version " << (data & FIT_PROTOCOL_VERSION_MAJOR_MASK) << FIT_PROTOCOL_VERSION_MAJOR_SHIFT << "." << (data & FIT_PROTOCOL_VERSION_MINOR_MASK) << " not supported.  Must be " << FIT_PROTOCOL_VERSION_MAJOR + ".15 or earlier.";
+                  message << "FIT decode error: Protocol version " << ((data & FIT_PROTOCOL_VERSION_MAJOR_MASK) >> FIT_PROTOCOL_VERSION_MAJOR_SHIFT) << "." << (data & FIT_PROTOCOL_VERSION_MINOR_MASK) << " not supported.  Must be " << FIT_PROTOCOL_VERSION_MAJOR << ".15 or earlier.";
                   throw RuntimeException(message.str());
                   return RETURN_ERROR;
                }
diff -ruN -x .svn fit/cpp/fit_encode.cpp libfit/cpp/fit_encode.cpp
--- fit/cpp/fit_encode.cpp	2012-02-01 20:24:36.000000000 -0800
+++ libfit/cpp/fit_encode.cpp	2012-04-18 11:17:15.897152678 -0700
@@ -13,7 +13,7 @@
 // Profile Version = 2.0Release
 // Tag = $Name: AKW2_000 $
 ////////////////////////////////////////////////////////////////////////////////
-
+#include <string.h>
 
 #include <iostream>
 #include "fit_encode.hpp"
diff -ruN -x .svn fit/cpp/fit_field.cpp libfit/cpp/fit_field.cpp
--- fit/cpp/fit_field.cpp	2012-02-01 20:24:36.000000000 -0800
+++ libfit/cpp/fit_field.cpp	2012-04-18 11:17:36.988654400 -0700
@@ -13,7 +13,7 @@
 // Profile Version = 2.0Release
 // Tag = $Name: AKW2_000 $
 ////////////////////////////////////////////////////////////////////////////////
-
+#include <string.h>
 
 #include <sstream>
 #include "fit_field.hpp"
diff -ruN -x .svn fit/cpp/Makefile libfit/cpp/Makefile
--- fit/cpp/Makefile	1969-12-31 16:00:00.000000000 -0800
+++ libfit/cpp/Makefile	2012-04-24 15:20:18.160216417 -0700
@@ -0,0 +1,24 @@
+
+SRC := fit_accumulated_field.cpp fit_accumulator.cpp fit_buffered_record_mesg_broadcaster.cpp fit_buffer_encode.cpp fit.cpp fit_crc.cpp fit_decode.cpp fit_encode.cpp fit_factory.cpp fit_field.cpp fit_field_definition.cpp fit_mesg_broadcaster.cpp fit_mesg.cpp fit_mesg_definition.cpp fit_mesg_with_event_broadcaster.cpp fit_profile.cpp fit_unicode.cpp
+
+OBJ := $(SRC:.cpp=.o)
+
+CXXFLAGS := -DFIT_USE_STDINT_H -fpic
+
+lib: libfit.a
+
+libfit.so: $(OBJ)
+	$(CXX) -fpic -shared $(OBJ) -o libfit.so
+
+libfit.a: $(OBJ)
+	$(AR) cr libfit.a $(OBJ)
+
+install:
+	/usr/bin/install -m 0644 libfit.a /usr/local/lib
+	/usr/bin/install -m 0755 -d /usr/include/libfit
+	/usr/bin/install -m 0644 *.hpp /usr/include/libfit
+
+clean:
+	$(RM) $(OBJ) libfit.so libfit.a
+
+
