diff -ruN libfit-15/cpp/BUILD libfit/cpp/BUILD
--- libfit-15/cpp/BUILD	1969-12-31 16:00:00.000000000 -0800
+++ libfit/cpp/BUILD	2015-06-20 12:35:49.999708033 -0700
@@ -0,0 +1,9 @@
+cc_library(
+  name = "libfit",
+  visibility = ["//visibility:public"],
+  srcs = glob(["fit*.cpp"]),
+  hdrs = glob(["fit*.cpp"]),
+  defines = [
+    "FIT_USE_STDINT_H"
+  ],
+)
diff -ruN libfit-15/cpp/fit_buffer_encode.cpp libfit/cpp/fit_buffer_encode.cpp
--- libfit-15/cpp/fit_buffer_encode.cpp	2015-06-02 18:11:20.000000000 -0700
+++ libfit/cpp/fit_buffer_encode.cpp	2015-06-20 11:45:32.811842552 -0700
@@ -15,6 +15,7 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 
+#include <string.h>
 #include "fit_buffer_encode.hpp"
 #include "fit_crc.hpp"
 
diff -ruN libfit-15/cpp/fit_decode.cpp libfit/cpp/fit_decode.cpp
--- libfit-15/cpp/fit_decode.cpp	2015-06-02 18:11:20.000000000 -0700
+++ libfit/cpp/fit_decode.cpp	2015-06-20 12:18:49.735753521 -0700
@@ -166,7 +166,7 @@
    state = STATE_FILE_HDR;
    lastTimeOffset = 0;
 
-   file.seekg(0, file.beg);
+   //file.seekg(0, file.beg);
    file.clear(); // workaround libc++ issue
 }
 
diff -ruN libfit-15/cpp/fit_encode.cpp libfit/cpp/fit_encode.cpp
--- libfit-15/cpp/fit_encode.cpp	2015-06-02 18:11:20.000000000 -0700
+++ libfit/cpp/fit_encode.cpp	2015-06-20 11:45:09.603843587 -0700
@@ -16,6 +16,7 @@
 
 
 #include <iostream>
+#include <string.h>
 #include "fit_encode.hpp"
 #include "fit_crc.hpp"
 
diff -ruN libfit-15/cpp/fit_field.cpp libfit/cpp/fit_field.cpp
--- libfit-15/cpp/fit_field.cpp	2015-06-02 18:11:20.000000000 -0700
+++ libfit/cpp/fit_field.cpp	2015-06-20 11:44:08.723846301 -0700
@@ -16,6 +16,7 @@
 
 
 #include <sstream>
+#include <string.h>
 #include "fit_field.hpp"
 #include "fit_mesg.hpp"
 #include "fit_unicode.hpp"
diff -ruN libfit-15/cpp/Makefile libfit/cpp/Makefile
--- libfit-15/cpp/Makefile	1969-12-31 16:00:00.000000000 -0800
+++ libfit/cpp/Makefile	2015-06-20 12:26:45.935732290 -0700
@@ -0,0 +1,23 @@
+
+SRC := $(wildcard *.cpp)
+OBJ := $(SRC:.cpp=.o)
+
+CXXFLAGS := -DFIT_USE_STDINT_H -fpic
+
+lib: libfit.a
+
+libfit.so: $(OBJ)
+	$(CXX) -fpic -shared $(OBJ) -o libfit.so
+
+libfit.a: $(OBJ)
+	$(AR) cr libfit.a $(OBJ)
+
+install:
+	/usr/bin/install -m 0644 libfit.a /usr/local/lib
+	/usr/bin/install -m 0755 -d /usr/include/libfit
+	/usr/bin/install -m 0644 *.hpp /usr/include/libfit
+
+clean:
+	$(RM) $(OBJ) libfit.so libfit.a
+
+
